src/detect-engine.c

```diff
@@ -2192,6 +2195,65 @@ uint8_t DetectEngineInspectBufferGeneric(DetectEngineCtx *de_ctx, DetectEngineTh
     const InspectionBuffer *buffer = engine->v2.GetData(det_ctx, transforms,
             f, flags, txv, list_id);
     if (unlikely(buffer == NULL)) {
+        if (eof) {
+            DetectContentData *cd;
```

Comment:
I do not follow you : `DetectContentData` is indeed built at parse-time. But it is what gets used at runtime in `DetectEngineContentInspectionInternal`...

Are you saying that this new logic should move to `DetectEngineAppInspectionEngine` ?  like a new field `match_on_null`... Why so ?

Reply:
Something like that ya. The idea is to avoid having this keyword specific logic in places where it doesn't really belong. It would also be another place to keep in sync with keyword additions and changes.

Also a large switch like this is inefficient, esp if we can precompute its outcome.

Reply:
See https://github.com/OISF/suricata/pull/10033

> It would also be another place to keep in sync with keyword additions and changes.

Well, this feature will exist somewhere in the code, and thus be "another place to keep in sync with keyword additions and changes" anyways, right ?

Reply:
Sure, but a better place than a match function. Besides, the match func isn't even used by all buffers.

Reply:
Doing that in `AppendAppInspectEngine`, only called by `DetectEngineAppInspectionEngine2Signature`

Reply:
I'm working towards consolidating all content inspection related to code in detect-engine-content-inspection.c, but since I'm making big changes there it's probably best not to modify it :)